evm/finality-block.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const {
+    providers: { JsonRpcProvider },
+} = ethers;
+const { Command, Option } = require('commander');
+const { printInfo, mainProcessor } = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+
+async function processCommand(config, chain, options) {
+    let { rpc, waitTime, update } = options;
+
+    if (!rpc) {
+        rpc = chain.rpc;
+    }
+
+    let max = 0;
+    let sum = 0;
+    const DELAY = 10000;
+
+    const provider = new JsonRpcProvider(rpc);
+
+    try {
+        await provider.getBlock('finalized');
+    } catch {
+        if (update && waitTime) {
+            chain.finality = parseInt(waitTime);
+        }
+
+        printInfo('Finalized tag not supported by rpc');
```

Comment:
Put the below checker in a different method. It can be called in the try block if finalized tag is supported otherwise you check the confirmations.